mongo/dataset_store.go

```diff
@@ -74,27 +74,48 @@ func (m *Mongo) Checker(ctx context.Context, state *healthcheck.CheckState) erro
 }
 
 // GetDatasets retrieves all dataset documents
-func (m *Mongo) GetDatasets(ctx context.Context) ([]models.DatasetUpdate, error) {
+func (m *Mongo) GetDatasets(ctx context.Context, offset, limit int, authorised bool) (*models.DatasetUpdateResults, error) {
 	s := m.Session.Copy()
 	defer s.Close()
 
-	iter := s.DB(m.Database).C("datasets").Find(nil).Iter()
+	var q *mgo.Query
+
+	if authorised {
+		q = s.DB(m.Database).C("datasets").Find(nil)
+	} else {
+		q = s.DB(m.Database).C("datasets").Find(bson.M{"current": bson.M{"$exists": true}})
+	}
+
+	iter := q.Sort().Skip(offset).Limit(limit).Iter()
+
 	defer func() {
 		err := iter.Close()
 		if err != nil {
 			log.Event(ctx, "error closing iterator", log.ERROR, log.Error(err))
 		}
 	}()
 
-	results := []models.DatasetUpdate{}
-	if err := iter.All(&results); err != nil {
+	totalCount, err := q.Count()
+	if err != nil {
+		log.Event(ctx, "error counting items", log.ERROR, log.Error(err))
```

Comment:
Think about returning here, otherwise when returning the data, `totalCount` will be incorrect and lead to poor data integrity

When handling an error, nearly always handle the error and return. Do not continue as this can lead to the application panicking. In this case the Count() function will always return 0 but I would advise returning error and handling the case as an internal server error (500)